packages/context/src/index.ts

```diff
@@ -119,3 +127,94 @@ export function MultiProvider<T extends readonly [unknown?, ...unknown[]]>(props
   };
   return fn(0);
 }
+
+/**
+ * A utility-function to provide context to components.
+ *
+ * @param children Accessor of Children
+ * @param context Context<T>
+ * @param value T
+ *
+ * @example
+ * ```tsx
+ * const NumberContext = createContext<number>
+ *
+ * const children = withContext(
+ *    () => props.children,
+ *    NumberContext,
+ *    1
+ * )
+ * ```
+ */
+
+export function withContext<T>(
+  children: Accessor<JSX.Element | JSX.Element[]>,
+  context: Context<T>,
+  value: T,
+) {
+  let result: JSX.Element | JSX.Element[];
+
+  context.Provider({
+    value,
+    children: (() => {
+      result = children();
+      return "";
+    }) as any as JSX.Element,
+  });
+
+  return () => result;
+}
+
+/*
+
+Type validation of the `values` array thanks to the amazing @otonashixav (https://github.com/otonashixav)
+
+*/
+
+/**
+ * A utility-function to provide multiple context to components.
+ *
+ * @param children Accessor of Children
+ * @param values Array of tuples of `[Context<T>, value T]`.
+ *
+ * @example
+ * ```tsx
+ * const NumberContext = createContext<number>
+ * const StringContext = createContext<string>
+ * const children = withContext(
+ *    () => props.children,
+ *    [
+ *      [NumberContext, 1],
+ *      [StringContext, "string"]
+ *    ]
+ * )
+ * ```
+ */
+
+export function withMultiContexts<T extends readonly [unknown?, ...unknown[]]>(
+  children: Accessor<JSX.Element | JSX.Element[]>,
+  values: {
+    [K in keyof T]: readonly [Context<T[K]>, [T[K]][T extends unknown ? 0 : never]];
+  },
+) {
+  let result: JSX.Element | JSX.Element[];
+
+  const fn = (index: number) => {
```

Comment:
Wouldn't it be simpler to use `values.reduce((children, [context, value]), children) => ...)`

Reply:
Something like
```ts
(values as [Context<any>, any]).reduce((acc, [context, value], index) => {
    return () =>
      context.Provider({
        value,
        children: () => {
          if (index === 0) result = acc();
          else acc();
        },
      });
  }, children)()
 ```
 ?
 
[playground](https://playground.solidjs.com/anonymous/c4747502-bcdb-4c5e-8c83-6fffad0b2856) 

I don't use `reduce` myself that much, so this code is a bit less readable to me personally, but that's mb more of a skill issue.